summingbird-storm/src/main/scala/com/twitter/summingbird/storm/builder/BaseBolt.scala

```diff
@@ -82,19 +81,10 @@ private[builder] case class BaseBolt[I, O](
   private[this] val rampPeriods = maxExecutePerSec.rampUptimeMS / PERIOD_LENGTH_MS
   private[this] val deltaPerPeriod: Long = if (rampPeriods > 0) (upperBound - lowerBound) / rampPeriods else 0
 
-  private[this] val outputFields: Option[Fields] =
-    outputEdges.headOption.map(_.edgeType.fields)
-  assert(
-    outputEdges.forall(edge => outputFields.map(_.toList) == Some(edge.edgeType.fields.toList)),
-    s"$boltId: Output edges should have same `Fields` $outputEdges")
+  private[this] val inputEdgesMap: Map[String, Topology.Edge[_, I]] =
```

Comment:
May be add a comment here as to why edge.source.id is unique so we don't end up overriding.